File: app/src/main/java/com/levnovikov/postbus/root/di/AppComponent.java
package com.levnovikov.postbus.root.di;

import com.levnovikov.core_profile.di.ProfileModule;
import com.levnovikov.postbus.Application;
import com.levnovikov.postbus.root.home.di.HomeComponentBuilder;

import dagger.Component;

/**
 * Author: lev.novikov
 * Date: 20/11/17.
 */

@RootScope
@Component(modules = {
        AppModule.class,
        ProfileModule.class,
        RootModule.class,
        HomeComponentBuilder.class })
public interface AppComponent {

    void inject(Application app);
}

File: app/src/main/java/com/levnovikov/postbus/root/di/AppComponent.kt
package com.levnovikov.postbus.root.di

import com.levnovikov.core_profile.di.ProfileModule
import com.levnovikov.postbus.root.home.di.HomeComponentBuilder
import com.levnovikov.system_base.base_di.ComponentBuilder
import dagger.Component
import javax.inject.Singleton

/**
 * Author: lev.novikov
 * Date: 20/11/17.
 */

@Singleton
@Component(modules = [(AppModule::class), (ProfileModule::class), (HomeComponentBuilder::class)])
interface AppComponent {

    fun subComponentBuilders(): Map<Class<out ComponentBuilder>, ComponentBuilder>
}
